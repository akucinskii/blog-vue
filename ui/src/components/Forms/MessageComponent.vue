<script setup lang="ts">
import { computed } from "vue";
const props = defineProps({
  text: {
    type: String,
    required: false,
    default: "",
  },
  type: {
    type: String,
    required: false,
    default: "success",
  },
});

const type = computed(() => {
  if (props.type === "error") {
    return "error";
  } else {
    return "success";
  }
});
</script>

<template>
  <v-row class="my-4">
    <v-alert id="message" :type="type" :color="type" icon="mdi-alert-circle-outline" close text-color="white" my="2">
      {{ props.text }}
    </v-alert>
  </v-row>
</template>
