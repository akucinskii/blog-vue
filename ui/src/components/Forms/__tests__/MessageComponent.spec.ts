import MessageComponent from "../MessageComponent.vue";
import { mount } from "@vue/test-utils";

describe("message component", () => {
  test("should match snapshot", () => {
    const wrapper = mount(MessageComponent);

    expect(wrapper.element).toMatchSnapshot();
  });

  test("should render the correct message", () => {
    const props = {
      text: "test message",
      type: "success",
    };

    const wrapper = mount(MessageComponent, {
      props,
    });
    expect(wrapper.find("#message").text()).toBe(props.text);
  });

  test("should render the success as default", () => {
    const props = {
      message: "test message",
      type: "SOMETHING",
    };

    const wrapper = mount(MessageComponent, {
      props,
    });

    expect(wrapper.vm.type).toBe("success");
  });

  test("should render the error type", () => {
    const props = {
      message: "test message",
      type: "error",
    };

    const wrapper = mount(MessageComponent, {
      props,
    });

    expect(wrapper.vm.type).toBe("error");
  });
});
