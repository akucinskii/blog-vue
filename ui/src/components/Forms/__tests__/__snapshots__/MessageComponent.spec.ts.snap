// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message component should match snapshot 1`] = `
<v-row
  class="my-4"
>
  <v-alert
    close=""
    color="success"
    icon="mdi-alert-circle-outline"
    id="message"
    my="2"
    text-color="white"
    type="success"
  >
    
  </v-alert>
</v-row>
`;
